src/main/java/com/iexec/blockchain/broker/BrokerService.java

```diff
@@ -26,87 +26,108 @@
 import com.iexec.common.sdk.order.payload.RequestOrder;
 import com.iexec.common.sdk.order.payload.WorkerpoolOrder;
 import com.iexec.common.utils.BytesUtils;
+import com.iexec.common.utils.FeignBuilder;
+import feign.Logger;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang3.StringUtils;
-import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
-import org.springframework.web.client.RestTemplate;
 
 import java.math.BigInteger;
+import java.text.MessageFormat;
+import java.util.Objects;
 import java.util.Optional;
 
 @Slf4j
 @Service
 public class BrokerService {
 
+    private final BrokerClient brokerClient;
     private final IexecHubService iexecHubService;
-    private final ChainConfig chainConfig;
 
 
     public BrokerService(ChainConfig chainConfig, IexecHubService iexecHubService) {
         //TODO Assert broker is up
-        this.chainConfig = chainConfig;
         this.iexecHubService = iexecHubService;
+        this.brokerClient = FeignBuilder.createBuilder(Logger.Level.BASIC)
+                .target(BrokerClient.class, chainConfig.getBrokerUrl());
     }
 
-    public String matchOrders(BrokerOrder brokerOrder) {
-        if (brokerOrder == null) {
-            throw new IllegalArgumentException("Broker order cannot be null");
+    void checkAssetOrder(String requestOrderAddress, String assetAddress, BigInteger assetPrice, String assetType) {
```

Comment:
Very minor: You can swap method ordering
from checkAssetOrder, checkRequestOrder, checkBrokerOrder
to checkBrokerOrder, checkRequestOrder, checkAssetOrder (calling order)

Reply:
done in 48b1921